src/query/interpret/eval.cpp

```diff
@@ -13,28 +13,42 @@
 
 namespace memgraph::query {
 
-int64_t EvaluateInt(ExpressionEvaluator *evaluator, Expression *expr, std::string_view what) {
-  TypedValue value = expr->Accept(*evaluator);
+int64_t EvaluateInt(ExpressionVisitor<TypedValue> &eval, Expression *expr, std::string_view what) {
```

Comment:
So I was a bit confused by the input types of both of these functions, so I changed them to be consistent. However, changing the return type is a bit tricky because I will be editing the code that's not in this PR. I can make a brief change in a different PR if that's okay so we separate concerns? EvaluateInt is currently not used in this PR's logic, but was used some time ago when evaluating bounds during path traversing (not part of this query)

Reply:
I agree with you that changes to `EvaluateInt` don't make sense in this PR, so let's do them in another one.